File: src/brawlify.py
from requests import get

class Brawlify:
	def __init__(self) -> None:
		self.api = "https://api.brawlapi.com"
		self.second_api = "https://proxy.brawlapi.com"
		self.headers = {
			"user-agent": "Brawlify.com/app"
		}

	def get_brawlers(self) -> dict:
		return get(
			f"{self.api}/v1/brawlers",
			headers=self.headers).json()

	def get_icons(self) -> dict:
		return get(
			f"{self.api}/v1/icons",
			headers=self.headers).json()

	def get_maps(self) -> dict:
		return get(
			f"{self.api}/v1/maps",
			headers=self.headers).json()

	def get_map_info(self, map_id: int) -> dict:
		return get(
			f"{self.api}/v1/maps/{map_id}",
			headers=self.headers).json()

	def get_game_modes(self) -> dict:
		return get(
			f"{self.api}/v1/gamemodes",
			headers=self.headers).json()

	def get_player_graphs(self, player_tag: str) -> dict:
		return get(
			f"{self.api}/v1/graphs/player/{player_tag}",
			headers=self.headers).json()

	def get_club_graphs(self, club_tag: str) -> dict:
		return get(
			f"{self.api}/v1/graphs/club/{club_tag}",
			headers=self.headers).json()

	def get_club_info(self, club_tag: str) -> dict:
		return get(
			f"{self.second_api}/v1/clubs/#{club_tag}",
			headers=self.headers).json()

	def get_club_logs(self, club_tag: str) -> dict:
		return get(
			f"{self.api}/v1/clublog/{club_tag}",
			headers=self.headers).json()
	
	def get_events(self) -> dict:
		return get(
			f"{self.api}/v1/events",
			headers=self.headers).json()

	def get_translations(self, language_code: str) -> dict:
		return get(
			f"{self.api}/v1/translations/{language_code}",
			headers=self.headers).json()
	
	def get_brawler_info(self, brawler_id: int) -> dict:
		return get(
			f"{self.api}/v1/brawlers/{brawler_id}",
			headers=self.headers).json()

File: src/brawlify.js
class Brawlify {
	constructor() {
		this.api = "https://api.brawlapi.com/v1"
		this.secondApi = "https://proxy.brawlapi.com"
		this.headers = {
			"User-Agent": "Brawlify.com/app"
		}
	}

	async getBrawlers() {
		const response = await fetch(
			`${this.api}/brawlers`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getIcons() {
		const response = await fetch(
			`${this.api}/icons`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getMaps() {
		const response = await fetch(
			`${this.api}/maps`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getGameModes() {
		const response = await fetch(
			`${this.api}/gamemodes`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getEvents() {
		const response = await fetch(
			`${this.api}/events`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getMapInfo(mapId) {
		const response = await fetch(
			`${this.api}/maps/${mapId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getPlayerGraphs(playerTag) {
		const response = await fetch(
			`${this.api}/graphs/player/${playerTag}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getClubGraphs(clubTag) {
		const response = await fetch(
			`${this.api}/graphs/club/${clubTag}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getClubInfo(clubTag) {
		const response = await fetch(
			`${this.api}/graphs/clubs/#${clubTag}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getClubLogs(clubTag) {
		const response = await fetch(
			`${this.api}/graphs/clublog/${clubTag}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}

	async getBrawlerInfo(brawlerId) {
		const response = await fetch(
			`${this.api}/brawlers/${brawlerId}`, {
				method: "GET",
				headers: this.headers
			})
		return response.json()
	}
}

module.exports = {Brawlify}
